src/Entity/EntityMeta.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\entity_meta_relation\Entity;
+
+use Drupal\Core\Entity\EntityChangedTrait;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\RevisionableContentEntityBase;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\entity_meta_relation\EntityMetaInterface;
+
+/**
+ * Defines the entity meta entity class.
+ *
+ * @ContentEntityType(
+ *   id = "entity_meta",
+ *   label = @Translation("Entity meta"),
+ *   label_collection = @Translation("Entity metas"),
+ *   bundle_label = @Translation("Entity meta type"),
+ *   handlers = {
+ *     "views_data" = "Drupal\views\EntityViewsData",
+ *     "form" = {
+ *       "add" = "Drupal\entity_meta_relation\Form\EntityMetaForm",
+ *       "edit" = "Drupal\entity_meta_relation\Form\EntityMetaForm",
+ *       "delete" = "Drupal\Core\Entity\ContentEntityDeleteForm"
+ *     },
+ *     "list_builder" = "Drupal\Core\Entity\EntityListBuilder",
+ *     "route_provider" = {
+ *       "html" = "Drupal\Core\Entity\Routing\AdminHtmlRouteProvider",
+ *     }
+ *   },
+ *   base_table = "entity_meta",
+ *   revision_table = "entity_meta_revision",
+ *   show_revision_ui = TRUE,
+ *   translatable = TRUE,
+ *   admin_permission = "administer entity meta types",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "revision" = "revision_id",
+ *     "bundle" = "bundle",
+ *     "label" = "id",
+ *     "langcode" = "langcode",
+ *     "uuid" = "uuid"
+ *   },
+ *   revision_metadata_keys = {
+ *     "revision_created" = "revision_timestamp",
+ *     "revision_log_message" = "revision_log"
+ *   },
+ *   links = {
+ *     "add-form" = "/admin/content/entity-meta/add/{entity_meta_type}",
```

Comment:
Since you made the `admin/structure/emr` route, let's put all these under the same path structure. Same for the other entity types.

Reply:
These routes are for content editing, not administration of the structure, should we move them there also?

Reply:
@hernani there will be no front-facing routes for edititng these entities, content or config. They serve an architectural purpose. You can argue we don't even need routes for EntityMeta entities. See content_moderation_state what they do. They dont even have forms somehow..